prometheus/src/main/java/org/opensearch/sql/prometheus/authinterceptors/AwsSigningInterceptor.java

```diff
@@ -48,11 +51,24 @@ public Response intercept(Interceptor.Chain chain) throws IOException {
     DateTimeFormatter timestampFormat = DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss'Z'")
         .withZone(ZoneId.of("GMT"));
 
-    Request newRequest = request.newBuilder()
+
+    Request.Builder newRequestBuilder = request.newBuilder()
         .addHeader("x-amz-date", timestampFormat.format(ZonedDateTime.now()))
-        .addHeader("host", request.url().host())
-        .build();
-    Request signed = okHttpAwsV4Signer.sign(newRequest, accessKey, secretKey);
+        .addHeader("host", request.url().host());
+
+    AWSCredentials awsCredentials = awsCredentialsProvider.getCredentials();
+    if (awsCredentialsProvider instanceof STSAssumeRoleSessionCredentialsProvider) {
+      newRequestBuilder.addHeader("x-amz-security-token",
+          ((STSAssumeRoleSessionCredentialsProvider) awsCredentialsProvider).getCredentials().getSessionToken());
+      LOG.info("x-amz-security-token: {}",
+          ((STSAssumeRoleSessionCredentialsProvider) awsCredentialsProvider).getCredentials().getSessionToken());
+    }
+    Request newRequest = newRequestBuilder.build();
+    LOG.info(awsCredentialsProvider.getClass());
+    LOG.info("Access Key: {}", awsCredentials.getAWSAccessKeyId());
+    LOG.info("Secret Key: {}", awsCredentials.getAWSSecretKey());
```

Comment:
Are you sure that you want to log your passwords as plain text?

Reply:
Thanks for pointing out...I was splitting a big PR and left these lines.